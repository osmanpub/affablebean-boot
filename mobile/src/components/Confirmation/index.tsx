import React from 'react';
import {Text, View} from 'react-native';
import {Order} from '../../interfaces/purchase';

type Props = {
  order: Order;
};

export default function Confirmation(props: Props) {
  const {orderRecord} = props.order;

  return (
    <View style={{padding: 16}}>
      <Text style={{fontWeight: 'bold'}} testID="confirm-success">
        Your order has been successfully processed and will be delivered within
        24 hours.
      </Text>
      <Text style={{paddingTop: 8}} testID="confirm-number">
        Please keep a note of your confirmation number:
        {orderRecord.confirmationNumber}
      </Text>
      <Text style={{paddingTop: 8}}>
        Thank you for shopping at the Affable Bean Green Grocer!
      </Text>
    </View>
  );
}
