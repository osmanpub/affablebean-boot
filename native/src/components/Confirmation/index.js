import React from "react";
import { Text, View } from "react-native";

export default function Confirmation(props) {
  const { orderRecord } = props.order;

  return (
    <View style={{ padding: 16 }}>
      <Text style={{ fontWeight: "bold" }}>
        Your order has been successfully processed and will be delivered within
        24 hours.
      </Text>
      <Text style={{ paddingTop: 8 }}>
        Please keep a note of your confirmation number:
        {orderRecord.confirmationNumber}
      </Text>
      <Text style={{ paddingTop: 8 }}>
        Thank you for shopping at the Affable Bean Green Grocer!
      </Text>
    </View>
  );
}
