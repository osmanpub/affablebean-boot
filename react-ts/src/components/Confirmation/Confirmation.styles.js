import styled from "styled-components";

export const ConfirmationText = styled.p`
  margin: 0 20px 20px;
  padding: 10px;
  background-color: #f5eabe;
  float: left;
  border-radius: 4px;
`;

export const DateProcessedRow = styled.td`
  padding: 2px 0 10px 20px;
`;

export const DeliveryAddressTable = styled.table`
  width: 70%;
  margin-left: 15%;
px;
`;

export const DeliveryAddressTableTd = styled.td`
  padding-top: 10px;
  padding-bottom: 10px;
`;

export const DeliverySurchargeCellLeft = styled.td`
  text-align: right;
  padding: 0;
`;

export const DeliverySurchargeCellRight = styled.td`
  padding: 0 20px;
`;

export const OrderSummaryTable = styled.table`
  width: 100%;
`;

export const TotalCellLeft = styled.td`
  text-align: right;
  padding: 0;
`;

export const TotalCellRight = styled.td`
  padding: 0 20px;
`;
