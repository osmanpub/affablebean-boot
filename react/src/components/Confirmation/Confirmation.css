.confirmationPriceColumn {
  width: 90px;
  font-weight: normal;
}

.summaryColumn {
  margin-top: 15px;
  width: 50%;
  float: left;
}
