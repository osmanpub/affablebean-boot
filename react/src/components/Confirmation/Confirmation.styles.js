import styled from "styled-components";

export const ConfirmationText = styled.p`
  margin: 0 20px 20px;
  padding: 10px;
  background-color: #f5eabe;
  float: left;
  border-radius: 4px;
`;
