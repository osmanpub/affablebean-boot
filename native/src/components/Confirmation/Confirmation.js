import React from "react";
import { Link } from "react-router-dom";
import {
  ConfirmationText,
  DateProcessedRow,
  DeliveryAddressTable,
  DeliveryAddressTableTd,
  DeliverySurchargeCellLeft,
  DeliverySurchargeCellRight,
  OrderSummaryTable,
  TotalCellLeft,
  TotalCellRight
} from "./Confirmation.styles";
import "./Confirmation.css";

export function Confirmation(props) {
  const { customer, orderedProducts, orderRecord, products } = props.order;

  const orderedProductsList = orderedProducts.map((product, index) => {
    const rowCol = index % 2 === 0 ? "white" : "lightBlue";
    const qty = product.quantity;
    const subtotal = (products[index].price * qty).toFixed(2);

    return (
      <tr className={`${rowCol}`} key={products[index].id}>
        <td>{products[index].name}</td>
        <td className="quantityColumn">{qty}</td>
        <td className="confirmationPriceColumn">&euro; {subtotal}</td>
      </tr>
    );
  });

  const tdStyle = {
    padding: "0 20px"
  };

  const surcharge = 3;

  return (
    <div className="singleColumn">
      <ConfirmationText>
        <strong>
          Your order has been successfully processed and will be delivered
          within 24 hours.
        </strong>
        <br />
        <br />
        Please keep a note of your confirmation number:
        <strong>{orderRecord.confirmationNumber}</strong>
        <br />
        If you have a query concerning your order, feel free to&nbsp;
        <Link to="/contact">contact us</Link>
        <br />
        <br />
        Thank you for shopping at the Affable Bean Green Grocer!
      </ConfirmationText>
      <div className="summaryColumn">
        <OrderSummaryTable className="detailsTable">
          <tbody>
            <tr className="header">
              <th colSpan={3}>order summary</th>
            </tr>
            <tr className="tableHeading">
              <td>product</td>
              <td>quantity</td>
              <td>price</td>
            </tr>
            {orderedProductsList}
            <tr className="lightBlue">
              <td colSpan={3} style={tdStyle}>
                <hr />
              </td>
            </tr>
            <tr className="lightBlue">
              <DeliverySurchargeCellLeft colSpan={2}>
                <strong>surcharge:</strong>
              </DeliverySurchargeCellLeft>
              <DeliverySurchargeCellRight>
                &euro; {surcharge.toFixed(2)}
              </DeliverySurchargeCellRight>
            </tr>
            <tr className="lightBlue">
              <TotalCellLeft colSpan={2}>
                <strong>total:</strong>
              </TotalCellLeft>
              <TotalCellRight>
                &euro; {orderRecord.amount.toFixed(2)}
              </TotalCellRight>
            </tr>
            <tr className="lightBlue">
              <td colSpan={3} style={tdStyle}>
                <hr />
              </td>
            </tr>
            <tr className="lightBlue">
              <DateProcessedRow colSpan={3}>
                <strong>date processed: </strong>
                <span>{orderRecord.dateCreated}</span>
              </DateProcessedRow>
            </tr>
          </tbody>
        </OrderSummaryTable>
      </div>
      <div className="summaryColumn">
        <DeliveryAddressTable className="detailsTable">
          <tbody>
            <tr className="header">
              <th colSpan={3}>delivery address</th>
            </tr>
            <tr>
              <DeliveryAddressTableTd colSpan={3} className="lightBlue">
                {customer.name}
                <br />
                {customer.address}
                <br />
                {customer.cityRegion}
                <br />
                <hr />
                {customer.email}
                <br />
                {customer.phone}
              </DeliveryAddressTableTd>
            </tr>
          </tbody>
        </DeliveryAddressTable>
      </div>
    </div>
  );
}
