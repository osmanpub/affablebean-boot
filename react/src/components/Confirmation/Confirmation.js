import React from "react";
import { Link } from "react-router-dom";
import { ConfirmationText } from "./Confirmation.styles";
import "./Confirmation.css";

export function Confirmation() {
  const { customer, orderedProducts, orderRecord, products } = this.props.order;

  return (
    <div className="singleColumn">
      <ConfirmationText>
        <strong>
          Your order has been successfully processed and will be delivered
          within 24 hours.
        </strong>
        <br />
        <br />
        Please keep a note of your confirmation number:
        <strong>{orderRecord.confirmationNumber}</strong>
        <br />
        If you have a query concerning your order, feel free to
        <Link to="/contact">contact us</Link>
        <a href="/contact">contact us</a>.
        <br />
        <br />
        Thank you for shopping at the Affable Bean Green Grocer!
      </ConfirmationText>
      <br />
    </div>
  );
}
