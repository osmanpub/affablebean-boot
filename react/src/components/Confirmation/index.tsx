import React from "react";
import { Link } from "react-router-dom";
import { getId } from "../../helpers/utils";
import { Order, OrderedProduct } from "../../interfaces/purchase";
import "./Confirmation.css";
import {
  ConfirmationText,
  DateProcessedRow,
  DeliveryAddressTable,
  DeliveryAddressTableTd,
  DeliverySurchargeCellLeft,
  DeliverySurchargeCellRight,
  OrderSummaryTable,
  TotalCellLeft,
  TotalCellRight
} from "./Confirmation.styles";

type Props = {
  order: Order;
};

export default function Confirmation(props: Props) {
  // BUG:for node, the orderedProducts and products arrays are not working!
  const { customer, orderedProducts, orderRecord, products } = props.order;

  const orderedProductsList = orderedProducts.map(
    (product: OrderedProduct, index: number) => {
      const rowCol = index % 2 === 0 ? "white" : "lightBlue";
      const qty = product.quantity;
      const subtotal = (products[index].price * qty).toFixed(2);

      return (
        <tr className={`${rowCol}`} key={getId(products[index])}>
          <td>{products[index].name}</td>
          <td className="quantityColumn">{qty}</td>
          <td className="confirmationPriceColumn">&euro; {subtotal}</td>
        </tr>
      );
    }
  );

  const tdStyle = {
    padding: "0 20px"
  };

  return (
    <div className="singleColumn">
      <ConfirmationText>
        <strong data-cy="confirm-success">
          Your order has been successfully processed and will be delivered
          within 24 hours.
        </strong>
        <br />
        <br />
        Please keep a note of your confirmation number:
        <strong>{orderRecord.confirmationNumber}</strong>
        <br />
        If you have a query concerning your order, feel free to&nbsp;
        <Link to="/contact">contact us</Link>
        <br />
        <br />
        Thank you for shopping at the Affable Bean Green Grocer!
      </ConfirmationText>
      <div className="summaryColumn">
        <OrderSummaryTable className="detailsTable">
          <tbody>
            <tr className="header">
              <th colSpan={3}>order summary</th>
            </tr>
            <tr className="tableHeading">
              <td>product</td>
              <td>quantity</td>
              <td>price</td>
            </tr>
            {orderedProductsList}
            <tr className="lightBlue">
              <td colSpan={3} style={tdStyle}>
                <hr />
              </td>
            </tr>
            <tr className="lightBlue">
              <DeliverySurchargeCellLeft colSpan={2}>
                <strong>surcharge:</strong>
              </DeliverySurchargeCellLeft>
              <DeliverySurchargeCellRight>
                &euro; {surcharge.toFixed(2)}
              </DeliverySurchargeCellRight>
            </tr>
            <tr className="lightBlue">
              <TotalCellLeft colSpan={2}>
                <strong>total:</strong>
              </TotalCellLeft>
              <TotalCellRight>
                &euro; {orderRecord.amount.toFixed(2)}
              </TotalCellRight>
            </tr>
            <tr className="lightBlue">
              <td colSpan={3} style={tdStyle}>
                <hr />
              </td>
            </tr>
            <tr className="lightBlue">
              <DateProcessedRow colSpan={3}>
                <strong>date processed: </strong>
                <span>{orderRecord.dateCreated}</span>
              </DateProcessedRow>
            </tr>
          </tbody>
        </OrderSummaryTable>
      </div>
      <div className="summaryColumn">
        <DeliveryAddressTable className="detailsTable">
          <tbody>
            <tr className="header">
              <th colSpan={3}>delivery address</th>
            </tr>
            <tr>
              <DeliveryAddressTableTd colSpan={3} className="lightBlue">
                {customer.name}
                <br />
                {customer.address}
                <br />
                {customer.cityRegion}
                <br />
                <hr />
                {customer.email}
                <br />
                {customer.phone}
              </DeliveryAddressTableTd>
            </tr>
          </tbody>
        </DeliveryAddressTable>
      </div>
    </div>
  );
}

const surcharge = 3;
