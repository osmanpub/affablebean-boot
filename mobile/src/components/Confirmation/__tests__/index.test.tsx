import React from 'react';
import renderer from 'react-test-renderer';
import Confirmation from '..';

const order = {
  customer: {
    id: 1,
    address: '11111111',
    ccNumber: '1111111111111111',
    cityRegion: 'NY',
    email: 'joe@bloggs.com',
    name: 'joe bloggs',
    phone: '11111111',
  },
  orderedProducts: [
    {
      quantity: 1,
    },
  ],
  orderRecord: {
    id: 1,
    amount: 5.39,
    confirmationNumber: '995604757',
    dateCreated: '1582123900751',
  },
  products: [
    {
      id: 12,
      description: 'contain peanuts<br>(3 cookies)',
      name: 'chocolate cookies',
      price: 2.39,
      category: {id: 3, name: 'bakery'},
    },
  ],
};

const confirmation = <Confirmation order={order} />;

describe('<Confirmation />', () => {
  it('renders correctly', () => {
    const component = renderer.create(confirmation).toJSON();
    expect(component).toMatchSnapshot();
  });
});
