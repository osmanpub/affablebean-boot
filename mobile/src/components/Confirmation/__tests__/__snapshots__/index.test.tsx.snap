// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Confirmation /> renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 16,
    }
  }
>
  <Text
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Your order has been successfully processed and will be delivered within 24 hours.
  </Text>
  <Text
    style={
      Object {
        "paddingTop": 8,
      }
    }
  >
    Please keep a note of your confirmation number:
    995604757
  </Text>
  <Text
    style={
      Object {
        "paddingTop": 8,
      }
    }
  >
    Thank you for shopping at the Affable Bean Green Grocer!
  </Text>
</View>
`;
